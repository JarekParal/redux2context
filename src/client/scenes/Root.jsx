/* @flow */
import React from "react";
import styled from "styled-components";

import { colors } from "../styles/variables";
import Text from "../components/Text";

const Div = styled.div`
  background: ${props => props.theme.colorPrimary};
`;

Div.defaultProps = {
  theme: {
    colorPrimary: colors.primary,
  },
};

const Root = () => (
  <Div>
    <Text t={__("Do you even lift?")} />
  </Div>
);

export default Root;
