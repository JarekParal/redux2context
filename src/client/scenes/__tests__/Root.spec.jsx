// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import Root from "../Root";

describe("#Root", () => {
  test("render", () => {
    const wrapper = shallow(<Root />);

    expect(wrapper).toMatchSnapshot();
  });
});
