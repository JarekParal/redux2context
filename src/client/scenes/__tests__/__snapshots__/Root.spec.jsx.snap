// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Root should render correctly 1`] = `
<Root__Div
  theme={
    Object {
      "colorPrimary": "deepskyblue",
    }
  }
>
  <WithIntl(Text)
    t="Do you even lift?"
  />
</Root__Div>
`;
